sdk/key_vault/src/certificate.rs

```diff
@@ -0,0 +1,816 @@
+    /// Runtime::new().unwrap().block_on(example());
+    /// ```
+    pub async fn restore_certificate(&mut self, backup_blob: &'a str) -> Result<(), Error> {
+        let mut uri = self.vault_url.clone();
+        uri.set_path("certificates/restore");
+        uri.set_query(Some(API_VERSION_PARAM));
+
+        let mut request_body = Map::new();
+        request_body.insert("value".to_owned(), Value::String(backup_blob.to_owned()));
+
+        self.post_authed(
+            uri.to_string(),
+            Some(Value::Object(request_body).to_string()),
+        )
+        .await?;
+
+        Ok(())
+    }
+
+    /// Restores a backed up certificate and all its versions.
+    /// This operation requires the certificates/restore permission.
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// use azure_key_vault::KeyClient;
+    /// use azure_identity::token_credentials::DefaultCredential;
+    /// use tokio::runtime::Runtime;
+    ///
+    /// async fn example() {
+    ///     let creds = DefaultCredential::default();
+    ///     let mut client = KeyClient::new(
+    ///         &"KEYVAULT_URL",
+    ///         &creds,
+    ///     ).unwrap();
+    ///     client.backup_certificate(&"CERTIFICATE_NAME").await.unwrap();
+    /// }
+    ///
+    /// Runtime::new().unwrap().block_on(example());
+    /// ```
+    pub async fn backup_certificate(
+        &mut self,
+        certificate_name: &'a str,
+    ) -> Result<KeyVaultCertificateBackupBlob, Error> {
+        let mut uri = self.vault_url.clone();
+        uri.set_path(&format!("certificates/{}/backup", certificate_name));
+        uri.set_query(Some(API_VERSION_PARAM));
+
+        let response_body = self.post_authed(uri.to_string(), None).await?;
+        let backup_blob = serde_json::from_str::<KeyVaultCertificateBackupResponseRaw>(
+            &response_body,
+        )
+        .map_err(|error| Error::BackupCertificateParseError {
+            error,
+            certificate_name: certificate_name.to_string(),
+            response_body,
+        })?;
+
+        Ok(KeyVaultCertificateBackupBlob {
+            value: backup_blob.value,
+        })
+    }
+
+    /// Deletes a certificate in the Key Vault.
+    ///
+    /// # Arguments
+    ///
+    /// * `certificate_name` - Name of the certificate
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// use azure_key_vault::{KeyClient, RecoveryLevel};
+    /// use azure_identity::token_credentials::DefaultCredential;
+    /// use tokio::runtime::Runtime;
+    ///
+    /// async fn example() {
+    ///     let creds = DefaultCredential::default();
+    ///     let mut client = KeyClient::new(
+    ///     &"KEYVAULT_URL",
+    ///     &creds,
+    ///     ).unwrap();
+    ///     client.delete_certificate(&"CERTIFICATE_NAME").await.unwrap();
+    /// }
+    ///
+    /// Runtime::new().unwrap().block_on(example());
+    /// ```
+    pub async fn delete_certificate(&mut self, certificate_name: &'a str) -> Result<(), Error> {
```

Comment:
This will actually need to return a long-running operation, but as with pageables, I think can be in a separate PR we can do later. I'll open an issue and would recommend adding a comment to reference the issue as a TODO.

Reply:
We can use https://github.com/Azure/azure-sdk-for-rust/issues/174 for tracking.